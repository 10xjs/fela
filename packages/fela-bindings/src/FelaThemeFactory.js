/* @flow */
import deprecate from './_deprecate'

export default function FelaThemeFactory(
  createElement: Function,
  ThemeContext: any
): any {
  function FelaTheme({ children, render }) {
    // TODO: remove with 11.0.0
    deprecate(
      render !== undefined,
      'The `render` prop in FelaTheme is deprecated. It will be removed in react-fela@11.0.0.\nPlease always use `children` instead. See http://fela.js.org/docs/api/bindings/fela-theme'
    )

    const renderFn = children || render

    return createElement(ThemeContext.Consumer, undefined, renderFn)
  }

  return FelaTheme
}
