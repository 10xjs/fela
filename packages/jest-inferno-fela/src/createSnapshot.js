import { render } from 'inferno'
import { createElement } from 'inferno-create-element'
import { createRenderer } from 'fela'
import { Provider, ThemeProvider } from 'inferno-fela'

import { createSnapshotFactory } from 'jest-fela-bindings'

export default createSnapshotFactory(
  createElement,
  render,
  createRenderer(),
  Provider,
  ThemeProvider
)
