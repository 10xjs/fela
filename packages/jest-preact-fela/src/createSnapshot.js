import { createElement, render } from 'preact'
import { createRenderer } from 'fela'
import { RendererProvider, ThemeProvider } from 'preact-fela'

import { createSnapshotFactory } from 'jest-fela-bindings'

export default createSnapshotFactory(
  createElement,
  render,
  createRenderer(),
  RendererProvider,
  ThemeProvider
)
