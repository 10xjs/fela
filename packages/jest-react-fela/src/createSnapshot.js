import { createElement } from 'react'
import { render } from 'react-dom'
import { createRenderer } from 'fela'
import { Provider, ThemeProvider } from 'react-fela'

import { createSnapshotFactory } from 'jest-fela-bindings'

export default createSnapshotFactory(
  createElement,
  render,
  createRenderer(),
  Provider,
  ThemeProvider
)
