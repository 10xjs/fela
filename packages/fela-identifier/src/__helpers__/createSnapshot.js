import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { render } from 'react-dom'
import { css } from 'js-beautify'

import RendererProviderFactory from '../../../fela-bindings/src/RendererProviderFactory'
import ThemeProviderFactory from '../../../fela-bindings/src/ThemeProviderFactory'
import { THEME_CHANNEL } from '../../../fela-bindings/src/themeChannel'

import renderToString from '../../../fela-tools/src/renderToString'
import createRenderer from '../../../fela/src/createRenderer'

const RendererProvider = RendererProviderFactory(
  Component,
  children => children,
  {
    childContextTypes: { renderer: PropTypes.object },
  }
)

const BaseThemeProvider = ThemeProviderFactory(
  Component,
  children => children,
  {
    propTypes: {
      theme: PropTypes.object.isRequired,
      overwrite: PropTypes.bool,
    },
    childContextTypes: {
      [THEME_CHANNEL]: PropTypes.object,
    },
    contextTypes: {
      [THEME_CHANNEL]: PropTypes.object,
    },
    defaultProps: {
      overwrite: false,
    },
  }
)

export default function createSnapshot(
  component,
  theme = {},
  renderer = createRenderer(),
  Provider = RendererProvider,
  ThemeProvider = BaseThemeProvider
) {
  const div = document.createElement('div')

  render(
    <Provider renderer={renderer}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </Provider>,
    div
  )

  return {
    html: div.innerHTML,
    css: css(renderToString(renderer)),
  }
}
