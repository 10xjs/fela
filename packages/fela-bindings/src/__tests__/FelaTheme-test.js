import 'raf/polyfill'
import React, { Component } from 'react'
import PropTypes from 'prop-types'

import FelaThemeFactory from '../FelaThemeFactory'
import { THEME_CHANNEL } from '../themeChannel'

import createSnapshot from '../__helpers__/createSnapshot'

const FelaTheme = FelaThemeFactory(Component, {
  [THEME_CHANNEL]: PropTypes.object,
})

describe('Using the FelaTheme component', () => {
  it('correctly pass the theme down', () => {
    expect(
      createSnapshot(
        <FelaTheme>
          {theme => <div>The color is {theme.color}.</div>}
        </FelaTheme>,
        { color: 'red' }
      )
    ).toMatchSnapshot()
  })
})
