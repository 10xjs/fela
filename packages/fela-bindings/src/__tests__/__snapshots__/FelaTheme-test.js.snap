// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the FelaTheme component correctly pass the theme down 1`] = `
Object {
  "css": "",
  "html": "<div>The color is red.</div>",
}
`;
